src/Serilog.Sinks.Loggly/Sinks/Loggly/FileBasedBookmarkProvider.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using Serilog.Debugging;
+
+namespace Serilog.Sinks.Loggly
+{
+    internal class FileBasedBookmarkProvider : IBookmarkProvider
+    {
+        readonly IFileSystemAdapter _fileSystemAdapter;
+        readonly Encoding _encoding;
+
+        Stream currentBookmarkFileStream;  
+        string _bookmarkFilename;
+
+        public FileBasedBookmarkProvider(string bufferBaseFilename, IFileSystemAdapter fileSystemAdapter, Encoding encoding)
+        {
+            _bookmarkFilename = Path.GetFullPath(bufferBaseFilename + ".bookmark");
+            _fileSystemAdapter = fileSystemAdapter;
+            _encoding = encoding;
+        }
+
+        public Bookmark GetCurrentBookmarkPosition()
+        {
+            EnsureCurrentBookmarkStreamIsOpen();
+
+            if (currentBookmarkFileStream.Length != 0)
+            {
+                using (var bookmarkStreamReader = new StreamReader(currentBookmarkFileStream, _encoding, false, 128, true))
+                {
+                    //set the position to 0, to begin reading the initial line
+                    bookmarkStreamReader.BaseStream.Position = 0;
+                    var bookmarkInfoLine = bookmarkStreamReader.ReadLine();
+
+                    if (bookmarkInfoLine != null)
+                    {
+                        //reset position after read
+                        var parts = bookmarkInfoLine.Split(new[] {":::"}, StringSplitOptions.RemoveEmptyEntries);
+                        if (parts.Length == 2)
+                        {
+                            return new Bookmark(long.Parse(parts[0]), parts[1]);
```

Comment:
please add error-handling for the case that the parsing fails

Reply:
Any suggestion for this,.  I've added (not yet pushed) a `long.TryParse` for the position value, but wondering if its worth adding anything for the path.